File: backend/Dtos/VerificationRequestDto.cs
using backend.Models;
using System.ComponentModel.DataAnnotations;

namespace backend.Dtos
{
  
    public class CreateVerificationRequestDto
    {
        [Required]
        public VerificationDocumentType DocumentType { get; set; }

        [Required]
        public string DocumentUrl { get; set; } = string.Empty;
    }

    public class AdminDecideVerificationRequestDto
    {
        [Required]
        public VerificationStatus Status { get; set; }

        [MaxLength(1000)]
        public string? AdminNote { get; set; }
    }

    public class VerificationRequestDto
    {
        public int Id { get; set; }
        public string UserId { get; set; } = string.Empty;
        public string FullName { get; set; } = string.Empty;
        public string UserName { get; set; } = string.Empty;
        public string? UserAvatarUrl { get; set; }
        public VerificationDocumentType DocumentType { get; set; }
        public string DocumentUrl { get; set; } = string.Empty;
        public VerificationStatus Status { get; set; }
        public string? AdminNote { get; set; }
        public string? ReviewedByAdminId { get; set; }
        public string? ReviewedByAdminName { get; set; }
        public string? ReviewedByAdminAvatarUrl { get; set; }
        public DateTime SubmittedAt { get; set; }
        public DateTime? ReviewedAt { get; set; }
    }

    //Lightweight list — for admin review queue
    public class VerificationRequestListDto
    {
        public int Id { get; set; }
        public string FullName { get; set; } = string.Empty;
        public string UserName { get; set; } = string.Empty;
        public string? UserAvatarUrl { get; set; }
        public VerificationDocumentType DocumentType { get; set; }
        public VerificationStatus Status { get; set; }
        public string? ReviewedByAdminName { get; set; }
        public DateTime SubmittedAt { get; set; }
        public DateTime? ReviewedAt { get; set; }
    }





    
}

File: frontend/src/app/dtos/verificationRequestDto.ts
import { VerificationDocumentType, VerificationStatus } from './enums';

export interface CreateVerificationRequestDto {
  documentType: VerificationDocumentType;
  documentUrl: string;
}

export interface AdminDecideVerificationRequestDto {
  status: VerificationStatus;
  adminNote?: string | null;
}

export interface VerificationRequestDto {
  id: number;
  userId: string;
  fullName: string;
  userName: string;
  userAvatarUrl: string | null;
  documentType: VerificationDocumentType;
  documentUrl: string;
  status: VerificationStatus;
  adminNote: string | null;
  reviewedByAdminId: string | null;
  reviewedByAdminName: string | null;
  reviewedByAdminAvatarUrl: string | null;
  submittedAt: string;
  reviewedAt: string | null;
}

export interface VerificationRequestListDto {
  id: number;
  fullName: string;
  userName: string;
  userAvatarUrl: string | null;
  documentType: VerificationDocumentType;
  status: VerificationStatus;
  reviewedByAdminName: string | null;
  submittedAt: string;
  reviewedAt: string | null;
}